### PR TITLE
[Mellanox] Add dpuctl services to support "dark mode".

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
#### Why I did it
Smart Switch Dark Mode allows disabling of all DPUs, removing any influence the DPU has on the system.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Add the `dpuctl.service` that runs on system start and shuts down DPUs if needed.

#### How to verify it
Compile the image and run it on the Nvidia Smart Switch. Verify that all DPUs are down after the system starts.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by Yakiv-Huryk — APPROVED** [review]

File: platform/mellanox/smartswitch/dpuctl/dpuctl.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# By default the dark mode is enabled
+DARK_MODE=true
+
+bf3_pci_id="15b3:c2d5"
+dpu2pcie[0]="06:00.0"
```
Comment (vvolam) [inline platform/mellanox/smartswitch/dpuctl/dpuctl.sh:23]: Is this static BDF? Will it be same for all HWSKU's of smartswitch?

File: platform/mellanox/smartswitch/dpuctl/dpuctl.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+dpu2pcie[2]="01:00.0"
+dpu2pcie[3]="02:00.0"
+
+if [[ -f /etc/mlnx/dpu.conf ]]; then
+    . /etc/mlnx/dpu.conf
+fi
+
+do_start() {
+    if [[ $DARK_MODE == "true" ]]; then
+        # By default all the DPUs are on. Power off the DPUs when is bark mode is required.
```
Comment (vvolam) [inline platform/mellanox/smartswitch/dpuctl/dpuctl.sh:34]: Nit: typo. It is "dark mode"!

File: platform/mellanox/smartswitch/dpuctl/dpuctl.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# By default the dark mode is enabled
+DARK_MODE=true
+
+bf3_pci_id="15b3:c2d5"
```
Comment (vvolam) [inline platform/mellanox/smartswitch/dpuctl/dpuctl.sh:22]: Just curious, Is this the vendor_id:device id of DPU?

File: platform/mellanox/smartswitch/dpuctl/dpuctl.sh (hunk 1 above)
Comment (oleksandrivantsiv) [reply platform/mellanox/smartswitch/dpuctl/dpuctl.sh:23]: Yes, the mapping will be the same.

File: platform/mellanox/smartswitch/dpuctl/dpuctl.sh (hunk 3 above)
Comment (oleksandrivantsiv) [reply platform/mellanox/smartswitch/dpuctl/dpuctl.sh:22]: Correct

**Review by vvolam — APPROVED** [review]
LGTM, Thanks!

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**